dev/reference/tools/data-migration/error-system.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Data Migration Error Message Reference
+summary: Learn the error message reference in Data Migration.
+category: reference
+---
+
+# Data Migration Error Message Reference
+
+This document describes the error messages output in Data Migration (DM) and offers solutions to some common error messages.
+
+## DM error system
+
+A new error system has been introduced in DM 1.0.0-GA, which has the following features:
+
++ Add the error code mechanism
++ Add the error fields such as `class`, `scope` or `level`
++ Improve the error description, error call chain information and stack trace information
+
+For the design and implementation of this error system, refer to [Proposal: Improve Error System](https://github.com/pingcap/dm/blob/master/docs/RFCS/20190722_error_handling.md).
+
+## Error message reference
+
+The following is an actual error message output in DM. Taking this message as a sample, this document explains each field of an error message in detail.
+
+```
+[code=38008:class=dm-master:scope=internal:level=high] grpc request error: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.17.0.2:8262: connect: connection refused"
+github.com/pingcap/dm/pkg/terror.(*Error).Delegate
+        /root/code/gopath/src/github.com/pingcap/dm/pkg/terror/terror.go:267
+github.com/pingcap/dm/dm/master/workerrpc.callRPC
+        /root/code/gopath/src/github.com/pingcap/dm/dm/master/workerrpc/rawgrpc.go:124
+github.com/pingcap/dm/dm/master/workerrpc.(*GRPCClient).SendRequest
+        /root/code/gopath/src/github.com/pingcap/dm/dm/master/workerrpc/rawgrpc.go:64
+github.com/pingcap/dm/dm/master.(*Server).getStatusFromWorkers.func2
+        /root/code/gopath/src/github.com/pingcap/dm/dm/master/server.go:1125
+github.com/pingcap/dm/dm/master.(*AgentPool).Emit
+        /root/code/gopath/src/github.com/pingcap/dm/dm/master/agent_pool.go:117
+runtime.goexit
+        /root/.gvm/gos/go1.12/src/runtime/asm_amd64.s:1337
+```
+
+All error messages in DM have the following three components:
+
++ [basic error information]
++ Error message description
++ Error stack information (optional)
+
+### Basic error information
+
+- `code`: error code, which is unique for each error type.
+
+    DM uses the same error code for the same error type. An error code does not change as DM version changes.
+
+    Some errors might be removed during the DM iteration, but the error code will not be removed. DM uses a new error code instead of an existing one for a new error type.
```

Comment:
```suggestion
    Some errors might be removed during the DM iteration, but the error code will not be removed. DM uses a new error code instead of an existing one for a new error.
```